### DISCUSSION TITLE
Exclusion by domain name (wildcard)

### DISCUSSION THREAD

**Discussion by presianbg** [Security-Onion-Solutions/securityonion#3342] [Unsupported Versions] [answered]
Hi,

First of all - THANK YOU for the AWESOME project! 

Is there a way to make an exclusion by domain name (wildcard preferably) ?
We have a lot of DNS traffic for a public domains we trust, our wish is to not  load the setup and flood the zeek logs with those domain queries.

I'm trying to create a BPF entry matching  the domain name with this cloudflare tool - https://github.com/cloudflare/bpftools , but can't figure out how to make it work with SecurityOnion 2.3 .

Cheers,
Presian

**Comment by TOoSmOotH** [comment, marked as answer]
You will need to write a zeek policy to not log specific domain names. Or if you are using Suricata as your meta data engine you can add filters based on the examples here: 
https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/idstools/sorules/filters.rules

**Reply by presianbg** [reply]
We use Zeek. 
Is the example for the suricata or for both ?
What is the recommended way (where) to write those filters ?

**Reply by TOoSmOotH** [reply]
That example is Suricata. 

Here are some zeek filter examples:

https://zeek.org/2012/02/02/filtering-logs-with-bro/

**Reply by presianbg** [reply]
Thank you! 

So the custom policy (script) should be something like this:

```zeek
module LogFilter;

event zeek_init()
{
        Log::remove_default_filter(DNS::LOG);
        Log::add_filter(DNS::LOG, [$name = "dns-noise",
                        $path_func(id: Log::ID, path: string, rec: DNS::Info) = {
                                return (rec?$query && /.mozilla.(org|net|com)$|.adobe.com$|.microsoft.com$|.local$|^localhost_prl$/ in rec$query) ? "dns-noise" : "dns";
                        }]);
}
```
So following the [docs](https://docs.securityonion.net/en/latest/zeek.html#custom-scripts):
 ```
Custom scripts can be added to /opt/so/conf/zeek/policy/custom/<$custom-module> on the manager.
```

But there is no such directory on the manager node... in fact there is no `zeek` directory in `/opt/so/conf/` at all.

So by looking where the other polices resides, the `custom` directory should be created in one of the following locations:
```
/opt/so/saltstack/default/salt/zeek/policy/
/opt/so/saltstack/local/salt/zeek/policy/
```
Which one is the correct ?

Cheers,
PY

**Reply by TOoSmOotH** [reply]
you want local

**Reply by presianbg** [reply]
Ok, after some reading and trial/error I made it work.

The example from 2012 is not optimal because it splits the log stream with `$path_func`. 
So if the pattern matches then it writes the content do different log file - `**dns-noise**.log` and if not in the `**dns**.log`.
Despite the dns-noise entries not being loaded in the ELK, this approach still consumes some disk I/O operations and space.


Here is what I used to remove the noise and not log it at all:

```zeek
module LogFilter;

event zeek_init()
{       
        Log::remove_default_filter(DNS::LOG);
        Log::add_filter(DNS::LOG, [$name = "dns-noisy", 
                        $pred(rec: DNS::Info) = {
                            if ((rec?$query) && (/.mozilla.(org|net|com)$|.microsoft.com$|.windowsupdate.com$|.local$/ in rec$query))
                                return F;
                            else
                                return T;
                            }]);
}
```
using the `$pred` is logging the event if the function return `True` and skipping the log if its `False`.

Not the most brilliant solution, but it works. Tested with DNS , HTTP and SSL.

Please let me know if there is more elegant way to achieve log filtering.
